src/ConnectionRequest.jl

```diff
@@ -34,13 +34,13 @@ end
 
 function getproxy(scheme, host)
     isnoproxy(host) && return nothing
-    if scheme == "http" && haskey(ENV, "http_proxy")
+    if scheme == "http" && haskey(ENV, "http_proxy") && !isempty(ENV["http_proxy"])
```

Comment:
```suggestion
    if scheme == "http" && !isempty(get(ENV, "http_proxy", ""))
```

You could do this with all of them, if you want (same behaviour)